### PR TITLE
new resource: azurerm_mssql_managed_instance_security_alert_policy

### PR DESCRIPTION
**Pull request by sinbai**
For test cases `
TestAccAzureRMMssqlManagedInstanceVulnerabilityAssessment_basic` and `
TestAccAzureRMMssqlManagedInstanceVulnerabilityAssessment_update` failed with "VulnerabilityAssessmentADSIsDisabled". To enable Advanced Data Security, Terraform should first support [Managed Server Security Alert Policies](https://github.com/Azure/azure-rest-api-specs/blob/6bf1d95b79b7ad55011bcc69dc6e887d5e002fa6/specification/sql/resource-manager/Microsoft.Sql/preview/2020-11-01-preview/ManagedServerSecurityAlertPolicies.json#L76). So I submitted this PR to support resource `azurerm_mssql_managed_instance_security_alert_policy` and enable Advanced Data Security for `azurerm_mssql_managed_instance_vulnerability_assessment` to fix test failure.

Note: 
Test case requires the env.ARM_TEST_LOCATION and env.ARM_TEST_LOCATION_ALT to be set to azure regional pair. E.g： if env.ARM_TEST_LOCATION is set to westcentralus and env.ARM_TEST_LOCATION_ALT is set to westus2, the test case could pass. 

Test Results:
image

### CODE REVIEW

File: website/docs/r/mssql_managed_instance_security_alert_policy.html.markdown [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `resource_group_name` - (Required) The name of the resource group that contains the MS SQL Managed Instance. Changing this forces a new resource to be created.
+
+* `managed_instance_name` - (Required) Specifies the name of the MS SQL Managed Instance. Changing this forces a new resource to be created.
+
+* `state` - (Required) Specifies the state of the policy, whether it is enabled or disabled or a policy has not been applied yet on the specific database. Possible values are `Disabled`, `Enabled`.
```
Comment (katbyte) [inline website/docs/r/mssql_managed_instance_security_alert_policy.html.markdown:226]: does this make sense to expose to the user? and should it not be a boolean called enabled?

File: website/docs/r/mssql_managed_instance_security_alert_policy.html.markdown [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+
+* `disabled_alerts` - (Optional) Specifies an array of alerts that are disabled. Possible values are `Sql_Injection`, `Sql_Injection_Vulnerability`, `Access_Anomaly`, `Data_Exfiltration`, `Unsafe_Action` and `Brute_Force`.
+
+* `email_account_admins` - (Optional) Boolean flag which specifies if the alert is sent to the account administrators or not. Defaults to `false`.
```
Comment (katbyte) [inline website/docs/r/mssql_managed_instance_security_alert_policy.html.markdown:230]:
```suggestion
* `email_account_admins_enabled` - (Optional) Boolean flag which specifies if the alert is sent to the account administrators or not. Defaults to `false`.
```

File: internal/services/mssql/mssql_managed_instance_security_alert_policy_resource.go [hunk 3]
```diff
@@ -0,0 +1,378 @@
...
+				ValidateFunc: validation.IntAtLeast(0),
+			},
+
+			"state": {
+				Type:     pluginsdk.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					string(sql.SecurityAlertPolicyStateDisabled),
+					string(sql.SecurityAlertPolicyStateEnabled),
+					string(sql.SecurityAlertPolicyStateNew),
```
Comment (katbyte) [inline internal/services/mssql/mssql_managed_instance_security_alert_policy_resource.go:93]: this doesn't seem like something a user would set? what happens if it is passed in?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/mssql_managed_instance_security_alert_policy.html.markdown (hunk 1 above)
Comment (sinbai) [reply website/docs/r/mssql_managed_instance_security_alert_policy.html.markdown]: @katbyte thanks for your feedback. Fixed and tested on TeamCity.
image

File: website/docs/r/mssql_managed_instance_security_alert_policy.html.markdown (hunk 2 above)
Comment (sinbai) [reply website/docs/r/mssql_managed_instance_security_alert_policy.html.markdown]: @katbyte  thanks for your feedback. Fixed.

File: internal/services/mssql/mssql_managed_instance_security_alert_policy_resource.go (hunk 3 above)
Comment (sinbai) [reply internal/services/mssql/mssql_managed_instance_security_alert_policy_resource.go]: @katbyte thanks for your feedback. Fixed.

**Review by katbyte — APPROVED** [review]
thanks @sinbai - LGTM 🏷️

**Pull request merged**